docs/sources/flow/reference/components/otelcol.connector.spanmetrics.md

```diff
@@ -259,42 +274,490 @@ otelcol.exporter.otlp "production" {
+                         }
+                       ],
+                       "startTimeUnixNano": "1702582936761872000",
+                       "timeUnixNano": "1702582936761872012",
+                       "asInt": "1"
+                     }
+                   ],
+                   "aggregationTemporality": 2,
+                   "isMonotonic": true
+                 }
+               }
+             ]
+           }
+         ]
+       },
+       {
+         "resource": {
+           "attributes": [
+             {
+               "key": "service.name",
+               "value": { "stringValue": "TestSvcName" }
+             },
+             {
+               "key": "k8s.pod.name",
+               "value": { "stringValue": "second" }
+             }
+           ]
+         },
+         "scopeMetrics": [
+           {
+             "scope": {
+               "name": "spanmetricsconnector"
+             },
+             "metrics": [
+               {
+                 "name": "calls",
+                 "sum": {
+                   "dataPoints": [
+                     {
+                       "attributes": [
+                         {
+                           "key": "service.name",
+                           "value": { "stringValue": "TestSvcName" }
+                         },
+                         {
+                           "key": "span.name",
+                           "value": { "stringValue": "TestSpan" }
+                         },
+                         {
+                           "key": "span.kind",
+                           "value": { "stringValue": "SPAN_KIND_UNSPECIFIED" }
+                         },
+                         {
+                           "key": "status.code",
+                           "value": { "stringValue": "STATUS_CODE_UNSET" }
+                         }
+                       ],
+                       "startTimeUnixNano": "1702582936761872000",
+                       "timeUnixNano": "1702582936761872012",
+                       "asInt": "1"
+                     }
+                   ],
+                   "aggregationTemporality": 2,
+                   "isMonotonic": true
+                 }
+               }
+             ]
+           }
+         ]
+       }
+     ]
+   }
+   ```
+   {{< /collapse >}}
+
+[prom-data-model]: https://prometheus.io/docs/concepts/data_model/
+
+Unfortunately, the [Prometheus data model][prom-data-model] has no notion of resource attributes.
+This means that if `otelcol.connector.spanmetrics` outputs metrics with identical metric attributes,
+but different resource attributes, `otelcol.exporter.prometheus` will convert the metrics into the same metric series.
+This problem can be solved by doing **either** of the following:
+
+- Recommended approach: Prior to `otelcol.connector.spanmetrics`, remove all resource attributes from the incoming spans which are not needed by `otelcol.connector.spanmetrics`.
```

Comment:
Thanks for writing this up and coming up with solutions to this problem. I worry that for users this all seems complex and the Agent is doing a poor job in simplifying this for the end user. 

Ideally, I would like to have a brief description of the problem and the ability to choose between the two recommended solutions using a one-line of config. This could be a module or a component that can hide this complexity away from the user. 

Can we consider putting this problem/solutions description in a different document as a short-term way to unblock users, but open an issue to build something with a better user experience?

Reply:
If there is another module/component which can abstract away these things, then that'd be great. However, this is the reference page for `otelcol.connector.spanmetrics`, so it feels like the right place to explain why it behaves the way it does and how to use it. 

I agree that we can try to be more straight to the point in this example section. I could make a new section outside of the Examples section where I explain the tendency to preserve resource attributes? And I could link to it from the Examples section, for users who want a deeper dive into why things are the way they are?

Reply:
> it feels like the right place to explain why it behaves the way it does

yeah, that certainly :) I just have doubts if this is the right place for what looks like a bit of a workaround... as mentioned, I'd prefer we had a way to address this with a single line of config, where we just say which of the two strategies to address the problem should be used. In practice it will be a few lines, of course, but we should perhaps consider a module or a change to the component in some way that will allow to resolve this problem in a more user-friendly way. 

Is there a way we can make this task simpler for the users?

Reply:
We could make it simpler for users if we offer a module which contains three components:
1. A transform processor which removes unnecessary resource attributes
1. A spanmetrics connector
1. An otelcol.prometheus.exporter

I could raise an issue for such a module to be created? This sort of thing is why modules are created, so I suppose we should be able to do it. But it'd be better wait and to do it with Modules v2 rather than with v1.

Reply:
Yeah, I agree with the module approach 👍 
Migration to a new module style shouldn't be too hard, but we can prioritise this independently nevertheless.